util/repo/util.go

```diff
@@ -0,0 +1,14 @@
+package repo
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// returns a formulated temporary directory location to clone a repository
+func TempRepoPath(repo string) string {
+	path := filepath.Join(os.TempDir(), strings.Replace(repo, "/", "_", -1))
+	_ = os.Mkdir(path, 0777)
```

Comment:
I think we should not create directories with such wide permissions as `0777`, even though we run in isolated containers. If no-one else beside user `argocd` needs to read/write from/to this repository, `0700` should be fine. Also, why not use `ioutil.Tempdir()`, which should guarantee a unique path.

Reply:
we don't want a unique path,  we want to re-use the same path